types/json.go

```diff
@@ -125,3 +126,42 @@ func UnmarshalJSON(b []byte, v *Value) error {
 	}
 	return nil
 }
+
+func unmarshalExtensionValue[T any](b []byte, extName string, parse func(string) (T, error)) (T, error) {
+	var zeroT T
+	var arg string
+	if len(b) > 0 && b[0] == '"' {
+		if err := json.Unmarshal(b, &arg); err != nil {
+			return zeroT, errors.Join(errJSONDecode, err)
+		}
+	} else {
+		var res extValueJSON
+		if err := json.Unmarshal(b, &res); err != nil {
+			return zeroT, errors.Join(errJSONDecode, err)
+		}
+		if res.Extn == nil {
+			// If we didn't find an Extn, maybe it's just an extn.
+			var res2 extn
+			_ = json.Unmarshal(b, &res2)
```

Comment:
Note to self: document why it's okay to ignore this error...

Reply:
Or not!